mw_strip_auth.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/TykTechnologies/logrus"
+)
+
+type MWStripAuthData struct {
+	*TykMiddleware
+}
+
+type MWStripAuthDataConfig struct{}
+
+// New lets you do any initialisations for the object can be done here
+func (m *MWStripAuthData) New() {}
+
+func (mw *MWStripAuthData) GetName() string {
+	return "MWStripAuthData"
+}
+
+// GetConfig retrieves the configuration from the API config - we user mapstructure for this for simplicity
+func (m *MWStripAuthData) GetConfig() (interface{}, error) {
+	var thisModuleConfig MiddlewareContextVarsConfig
+	return thisModuleConfig, nil
+}
+
+func (a *MWStripAuthData) IsEnabledForSpec() bool {
+	if a.Spec.StripAuthData {
+		return true
+	}
+	return false
+}
+
+func (m *MWStripAuthData) StripAuth(r *http.Request, spec *APISpec) {
+	if spec.APIDefinition.Auth.UseParam {
+		// Remove the query string value
+		cp, err := url.Parse(r.URL.String())
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"path":   r.URL.Path,
+				"origin": GetIPFromRequest(r),
+				"key":    "",
+			}).Error("Failed to copy URL to strip auth: ", err)
+			return
+		}
+
+		q := cp.Query()
+		n := spec.APIDefinition.Auth.ParamName
+		if n == "" {
+			n = spec.APIDefinition.Auth.AuthHeaderName
+		}
+		q.Del(n)
+		cp.RawQuery = q.Encode()
+		r.URL, err = r.URL.Parse(cp.String())
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"path":   r.URL.Path,
+				"origin": GetIPFromRequest(r),
+				"key":    "",
+			}).Error("Failed to set new URL: ", err)
+			return
+		}
+	}
+
+	n := spec.APIDefinition.Auth.AuthHeaderName
+	if n == "" {
+		n = "Authorization"
+	}
+
+	key := r.Header.Get(n)
+	if key != "" {
+		r.Header.Del(n)
+	}
+}
+
+// ProcessRequest will run any checks on the request on the way through the system, return an error to have the chain fail
+func (m *MWStripAuthData) ProcessRequest(w http.ResponseWriter, r *http.Request, configuration interface{}) (error, int) {
+
+	if !m.Spec.StripAuthData {
```

Comment:
nit

```
if m.Spec.StripAuthData {
    m.StripAuth(r, mSpec)
}

return nil, 200
```

Also, does ProcessRequest function implement an interface and that's why we are also returning an error?

Reply:
I wouldn't worry about nits at all, as this patch will be completely different once ported to master :)

And yes, this is an interface.